plugins/database/quota/src/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```diff
@@ -0,0 +1,419 @@
+        response.setCurrency(QuotaConfig.QuotaCurrencySymbol.value());
+        return response;
+    }
+
+    private QuotaEmailTemplateResponse createQuotaEmailResponse(QuotaEmailTemplatesVO template) {
+        QuotaEmailTemplateResponse response = new QuotaEmailTemplateResponse();
+        response.setTemplateType(template.getTemplateName());
+        response.setTemplateSubject(template.getTemplateSubject());
+        response.setTemplateText(template.getTemplateBody());
+        response.setLocale(template.getLocale());
+        response.setLastUpdatedOn(template.getLastUpdated());
+        return response;
+    }
+
+    @Override
+    public List<QuotaEmailTemplateResponse> listQuotaEmailTemplates(QuotaEmailTemplateListCmd cmd) {
+        final String templateName = cmd.getTemplateName();
+        List<QuotaEmailTemplatesVO> templates = _quotaEmailTemplateDao.listAllQuotaEmailTemplates(templateName);
+        final List<QuotaEmailTemplateResponse> responses = new ArrayList<QuotaEmailTemplateResponse>();
+        for (final QuotaEmailTemplatesVO template : templates) {
+            responses.add(createQuotaEmailResponse(template));
+        }
+        return responses;
+    }
+
+    @Override
+    public boolean updateQuotaEmailTemplate(QuotaEmailTemplateUpdateCmd cmd) {
+        final String templateName = cmd.getTemplateName();
+        final String templateSubject = StringEscapeUtils.escapeJavaScript(cmd.getTemplateSubject());
+        final String templateBody = StringEscapeUtils.escapeJavaScript(cmd.getTemplateBody());
+        final String locale = cmd.getLocale();
+
+        final List<QuotaEmailTemplatesVO> templates = _quotaEmailTemplateDao.listAllQuotaEmailTemplates(templateName);
+        if (templates.size() == 1) {
+            final QuotaEmailTemplatesVO template = templates.get(0);
+            template.setTemplateSubject(templateSubject);
+            template.setTemplateBody(templateBody);
+            if (locale != null) {
+                template.setLocale(locale);
+            }
+            return _quotaEmailTemplateDao.updateQuotaEmailTemplate(template);
+        }
+        return false;
+    }
+
+    @Override
+    public QuotaBalanceResponse createQuotaLastBalanceResponse(List<QuotaBalanceVO> quotaBalance, Date startDate) {
+        if (quotaBalance.size() == 0) {
+            new InvalidParameterValueException("There are no balance entries on or before the requested date.");
+        }
+        if (startDate == null) {
+            startDate = new Date();
+        }
+        QuotaBalanceResponse resp = new QuotaBalanceResponse();
+        BigDecimal lastCredits = new BigDecimal(0);
+        for (Iterator<QuotaBalanceVO> it = quotaBalance.iterator(); it.hasNext();) {
+            QuotaBalanceVO entry = it.next();
+            s_logger.info("createQuotaLastBalanceResponse Date=" + entry.getUpdatedOn() + " balance=" + entry.getCreditBalance() + " credit=" + entry.getCreditsId());
```

Comment:
Since this is the first release the debug messages have been left purposefully there in order to assist in debugging. These will be removed in later version.